### PR TITLE
[FIX] web_editor: deleteforward before contenteditable

### PR DESCRIPTION
**Pull request by dhba-odoo**
**Current behavior before PR:**

When an elements next sibling is a whitespace node, followed by a `contentEditable=false` element, performing the deleteforward action did not remove the `contentEditable=false` element.

**Desired behavior after PR is merged:**

When an elements next sibling is a whitespace node, followed by a `contentEditable=false` element, performing the deleteforward action will now remove both the whitespace and the entire `contentEditable=false` element.

task:4058770

### CODE REVIEW

File: addons/web_editor/static/src/js/editor/odoo-editor/src/commands/deleteForward.js [hunk 1]
```diff
@@ -188,6 +188,14 @@ HTMLElement.prototype.oDeleteForward = function (offset) {
         return;
     }
 
+    if (nextSibling && isWhitespace(nextSibling)) {
+        // If the next sibling is a whitespace, remove it.
+        nextSibling.remove();
+        const parent = this.parentElement;
+        const index = childNodeIndex(this);
+        this.oDeleteForward(parent, index);
+        return;
```
Comment (deso-odoo) [inline addons/web_editor/static/src/js/editor/odoo-editor/src/commands/deleteForward.js:197]: Why not just do
```js
nextSibling.remove();
nextSibling = this.nextSibling;
```
where nextSibling is declared?

Also test case is missing.

File: addons/web_editor/static/src/js/editor/odoo-editor/test/spec/editor.test.js [hunk 2]
```diff
@@ -490,6 +490,15 @@ X[]
                         contentAfter: `<p>[]&nbsp;def</p>`,
                     });
                 });
+                it('should remove the contentEditable false element', async () => {
+                    await testEditor(BasicEditor, {
+                        contentBefore: `<p>test[]</p>\n<div contentEditable="false"><p>abc</p></div>`,
```
Comment (dmo-odoo) [inline addons/web_editor/static/src/js/editor/odoo-editor/test/spec/editor.test.js:495]: Could you add some content after the `contenteditable=false` div please? So that the test can check that whatever was after it is still present at the end of the test.

File: addons/web_editor/static/src/js/editor/odoo-editor/src/commands/deleteForward.js [hunk 3]
```diff
@@ -163,7 +163,13 @@ HTMLElement.prototype.oDeleteForward = function (offset) {
...
     }
 
-    const nextSibling = this.nextSibling;
+    let nextSibling = this.nextSibling;
+    while (nextSibling && isWhitespace(nextSibling)) {
+        // If the next sibling is a whitespace, remove it.
+        nextSibling.remove();
+        nextSibling = this.nextSibling;
+    }
+
```
Comment (dmo-odoo) [inline addons/web_editor/static/src/js/editor/odoo-editor/src/commands/deleteForward.js:172]: I'm kind of confused as to why this change might fix the bug? I mean the change only affects whitespace but somehow fixes the behavior around contenteditables? How? ^^'

Comment (dhba-odoo) [reply addons/web_editor/static/src/js/editor/odoo-editor/src/commands/deleteForward.js:172]: The issue produce when the html structure is like: `<p><br>[]</p>\n<div contentEditable="false"></div>`
In the deleteForward, when we find nextsibling of this using `this.nextSibling` it will be a whitespace. As a result, the code that remove `contentEditable=false` elements does not execute, as the next sibling is not a `contentEditable=false` element.

so removing that whitespce before the code that remove `contentEditable=false` elements is executed fix the issue.

Comment (dmo-odoo) [reply addons/web_editor/static/src/js/editor/odoo-editor/src/commands/deleteForward.js:172]: That seems too broad a fix, I mean I fear we might be deleting some whitespace that has useful value at some point in the future in some edge case due to that code. Could you maybe add a check that the whitespace has a cType of BLOCK_OUTSIDE on both sides? I think that would ensure that it is in fact not rendered. I'm not sure so let me know if it works.

Comment (dhba-odoo) [reply addons/web_editor/static/src/js/editor/odoo-editor/src/commands/deleteForward.js]: @dmo-odoo i have applied the requested changes and its working

**Comment by dmo-odoo** [comment]
@dhba-odoo Thanks!
@robodoo r+

File: addons/web_editor/static/src/js/editor/odoo-editor/src/commands/deleteForward.js [hunk 4]
```diff
@@ -163,7 +163,18 @@ HTMLElement.prototype.oDeleteForward = function (offset) {
...
+    while (nextSibling && isWhitespace(nextSibling)) {
+        const index = childNodeIndex(nextSibling);
+        const left = getState(nextSibling, index, DIRECTIONS.LEFT).cType;
+        const right = getState(nextSibling, index, DIRECTIONS.RIGHT).cType;
+        if (left === CTYPES.BLOCK_OUTSIDE && right === CTYPES.BLOCK_OUTSIDE) {
+            // If the next sibling is a whitespace, remove it.
+            nextSibling.remove();
+            nextSibling = this.nextSibling;
+        }
+    }
```
Comment (deso-odoo) [inline addons/web_editor/static/src/js/editor/odoo-editor/src/commands/deleteForward.js:176]: Saw the changes just now won't this need an `break` in else?

Comment (dmo-odoo) [reply addons/web_editor/static/src/js/editor/odoo-editor/src/commands/deleteForward.js:176]: I think so, yes. And we'll need a test for it as well because somehow it wasn't caught by any of our current tests.

Comment (dhba-odoo) [reply addons/web_editor/static/src/js/editor/odoo-editor/src/commands/deleteForward.js:176]: @dmo-odoo oops, I overlooked such case! I have made a new [PR](https://github.com/odoo/odoo/pull/184557) with a test case for it. Please have a look.
@deso-odoo  Thanks for pointing it out.
